pkg/auth/providers/saml/saml_client.go

```diff
@@ -363,6 +363,9 @@ func (s *Provider) HandleSamlAssertion(w http.ResponseWriter, r *http.Request, a
 	rancherAction := s.clientState.GetState(r, "Rancher_Action")
 
 	if rancherAction == loginAction {
+		if !strings.Contains(redirectURL, "/dashboard/auth") {
```

Comment:
Could you add `/dashboard/auth` and `/login?` as constants?

Reply:
Okay